assemblyline/odm/models/config.py

```diff
@@ -282,6 +282,31 @@ class OAuth(odm.Model):
 }
 
 
+DEFAULT_SAML_PROVIDERS = {}
+
+
+@odm.model(index=False, store=False, description="SAML Configuration")
+class SAML(odm.Model):
+    enabled: bool = odm.Boolean(description="Enable use of SAML?")
+    config_dir: str = odm.Keyword(description="SAML config directory")
+    auto_create: bool = odm.Boolean(description="Auto-create users if they are missing")
+    auto_sync: bool = odm.Boolean(description="Should we automatically sync with SAML server on each login?")
+    email_attribute_name: str = odm.Keyword(description="SAML attribute name for a user's email address ")
+    first_name_attribute_name: str = odm.Keyword(description="SAML attribute name for a user's first name")
+    last_name_attribute_name: str = odm.Keyword(description="SAML attribute name for a user's last name")
+
+
+DEFAULT_SAML = {
+    "enabled": True,
```

Comment:
Probably a good idea to set the default for `enabled: False` in cases where deployments don't have SAML auth setup.